### PR TITLE
add bert_hf openwebtext

### PR DESCRIPTION
**Pull request by shh2000**
completely fp32/fp16/amp, no apex, no torch.cuda, no pycuda
MFU = 25.9, no compile(if nvidia use pytorch2.0 with TorchInductor, it would increase to 30.0% or higher)
use small but easy-get dataset(32MiB, but enough for performance testing). Maybe change into formal dataset(like 1.0GiB or bigger) later

### CODE REVIEW

**Review by yuzhou03 — APPROVED** [review]

**Comment by yuzhou03** [comment]
test_conf中，添加一下运行配置示例

File: training/nvidia/bert_hf-pytorch/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+| 训练吞吐量     | p_train,见“性能指标”    | 不包含每个epoch末尾的评估部分耗时     |
+| **计算吞吐量** | **p_core,见“性能指标”** | 不包含数据IO部分的耗时      |
+| 训练结果       | acc,见“性能指标”        | masked_lm任务准确率(实际/目标) |
+| 额外修改项     | fp16实现方式 | nvidia使用1+8+7格式(bf16)来实现16位浮点数 |
+
+* 性能指标
+
+| 配置               | precision | fix_hp | e2e_time | p_whole | p_train | p_core | acc  | mem |
+| ------------------ | --------- | ---- | ---- | ---- | ---- | ---- |  ---- | ---- |
+| A100单机8卡（1x8） | fp32 | bs=12 | 173.7    | 258.7   | 277.1   | 295.6  | 0.685/0.68 | 31.2/40.0 |
```
Comment (upvenly) [inline training/nvidia/bert_hf-pytorch/README.md:41]: 训练时间太短了

**Comment by yuzhou03** [comment]
解决下代码冲突

File: training/nvidia/bert_hf-pytorch/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+| **计算吞吐量** | **p_core,见“性能指标”** | 不包含数据IO部分的耗时      |
+| 训练结果       | acc,见“性能指标”        | masked_lm任务准确率(实际/目标) |
+| 额外修改项     | fp16实现方式 | nvidia使用1+8+7格式(bf16)来实现16位浮点数 |
+
+* 性能指标
+
+| 配置               | precision | fix_hp | e2e_time | p_whole | p_train | p_core | acc  | mem |
+| ------------------ | --------- | ---- | ---- | ---- | ---- | ---- |  ---- | ---- |
+| A100单机8卡（1x8） | fp32 | bs=12 | 2133.7 | 285.4 | 294.5 | 299.3 | 0.657/0.655 | 31.2/40.0 |
+| A100两机8卡（1x1） | fp32 | bs=12 |  |  |     |  | |  |
```
Comment (yuzhou03) [inline training/nvidia/bert_hf-pytorch/README.md:42]: A100单机单卡（1x1)

Comment (yuzhou03) [reply training/nvidia/bert_hf-pytorch/README.md:42]: 请补充1x1和2x8数据

**Comment by yuzhou03** [comment]
代码有冲突

**Review by upvenly — APPROVED** [review]

**Pull request merged**